### PR TITLE
chore(botocore): refactor botocore integration

### PR DESCRIPTION
**Pull request by wconti27**
Refactors botocore integration and separates contrib into files by service. In preparation for further botocore changes in: https://github.com/DataDog/dd-trace-py/pull/7762

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/contrib/botocore/services/sqs.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+from ..utils import get_queue_name
+from ..utils import set_patched_api_call_span_tags
+from ..utils import set_response_metadata_tags
+
+
+log = get_logger(__name__)
+
+
+def _encode_data(trace_data):
+    """
```
Comment (mabdinur) [inline ddtrace/contrib/botocore/services/sqs.py:32]: nit: We should add typing to all methods. Honestly this should be a ci check

File: ddtrace/contrib/botocore/services/sqs.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    return json.dumps(trace_data)
+
+
+def get_topic_arn(params):
+    # type: (str) -> str
+    """
+    :params: contains the params for the current botocore action
+    Return the name of the topic given the params
+    """
+    sns_arn = params["TopicArn"]
```
Comment (mabdinur) [inline ddtrace/contrib/botocore/services/sqs.py:45]: Is topic arn guarenteed to exist? 

```suggestion
    sns_arn = params.get("TopicArn")
```

File: ddtrace/contrib/botocore/services/sqs.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+            # Use String since changing this to Binary would be a breaking
+            # change as other tracers expect this to be a String.
+            entry["MessageAttributes"]["_datadog"] = {"DataType": "String", "StringValue": _encode_data(trace_data)}
+        elif endpoint_service == "sns":
+            # Use Binary since SNS subscription filter policies fail silently
+            # with JSON strings https://github.com/DataDog/datadog-lambda-js/pull/269
+            # AWS will encode our value if it sees "Binary"
+            entry["MessageAttributes"]["_datadog"] = {"DataType": "Binary", "BinaryValue": _encode_data(trace_data)}
+        else:
+            log.warning("skipping trace injection, endpoint is not SNS or SQS")
```
Comment (mabdinur) [inline ddtrace/contrib/botocore/services/sqs.py:71]: Can these logs be noisy? Should we make them debug logs or is it a sign something is very wrong

File: ddtrace/contrib/botocore/services/sqs.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    Inject trace headers and DSM info into MessageAttributes for all SQS or SNS records inside a batch
+    """
+
+    trace_data = {}
+    HTTPPropagator.inject(span.context, trace_data)
+
+    # An entry here is an SNS or SQS record, and depending on how it was published,
+    # it could either show up under Entries (in case of PutRecords),
+    # or PublishBatchRequestEntries (in case of PublishBatch).
+    entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
```
Comment (mabdinur) [inline ddtrace/contrib/botocore/services/sqs.py:94]: Is entries ever expected to be empty? Do we need a warning here?

**Review by mabdinur — APPROVED** [review]
left a few nits and questions. Overall LGTM

File: ddtrace/contrib/botocore/services/sqs.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    operation = function_vars.get("operation")
+
+    with pin.tracer.trace(
+        trace_operation,
+        service=schematize_service_name("{}.{}".format(pin.service, endpoint_name)),
+        span_type=SpanTypes.HTTP,
+    ) as span:
+        set_patched_api_call_span_tags(span, instance, args, params, endpoint_name, operation)
+
+        if args:
```
[the remark this answers is not in the document]
Comment (wconti27) [reply ddtrace/contrib/botocore/services/sqs.py:145]: removed the `if args` check. This was in place previously to check whether to get argument values like operation

File: ddtrace/contrib/botocore/services/sqs.py (hunk 1 above)
Comment (wconti27) [reply ddtrace/contrib/botocore/services/sqs.py:32]: Maybe next PR! 😆

**Pull request merged**